### PR TITLE
fix: reproduce symlinks in build staging correctly on windows

### PR DESCRIPTION
**Pull request by stefreak**
**What this PR does / why we need it**:
Unfortunately I mixed those up due to the confusing data structures in https://github.com/garden-io/garden/pull/6430

### CODE REVIEW

File: core/src/build-staging/helpers.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ function doClone(params: CopyParams) {
           to,
           // relevant on windows
           // nodejs will auto-detect this on windows, but if the symlink is copied before the target then the auto-detection will get it wrong.
-          targetStats?.isDirectory() ? "dir" : "file",
+          sourceStats.target?.isDirectory() ? "dir" : "file",
```
Comment (stefreak) [inline core/src/build-staging/helpers.ts:186]: probably would have been less confusing if these parameters were named `fromStats` and `toStats` or something like that

Comment (vvagaytsev) [reply core/src/build-staging/helpers.ts:186]: Feel free to rename params to avoid confusion.

**Review by vvagaytsev — CHANGES_REQUESTED** [review]
There is a lint error. It complains about unused local variable `targetStats` from the params object defined in [L159](https://github.com/garden-io/garden/pull/6433/files#diff-35d7acac46ec199eb991f6b1296b6ae1d7a11ec53e57668afcb91a194a9cea00R159).

Do we need that variable here?

**Comment by stefreak** [comment]
@vvagaytsev thanks, fixed the issue now, and refactored the code to make the variable names more clear.

**Comment by stefreak** [comment]
@vvagaytsev the parameter has been added in #6430, but it was only due to the misleading name; It was actually correct to use `sourceStats.target` so the `sourceStats` parameter is indeed not necessary.

**Review by vvagaytsev — APPROVED** [review]
Thanks! 👍

**Pull request merged**